rsmetrics.py

```diff
@@ -90,18 +92,21 @@ def inner():
     run.recommendations=run.recommendations[run.recommendations['Timestamp'] < args.endtime]
 
 
+
 # users are populated with two columns: one includes user id and the other includes a list of accessed services
 run.users=pd.read_csv(os.path.join(args.input,'users.csv'),names=["User","Services"],converters={'Services': lambda x: map(int,x.split())})
     
 # populate services
 if args.services:
     run.services=pd.read_csv(os.path.join(args.input,'services.csv'),names=["Service"])
 
+# In some metrics we need user actions that might have targeted urls with services that no longer exist
+# 
 # remove user actions when user or service does not exist in users' or services' catalogs
 # adding -1 in all catalogs indicating the anonynoums users or not-known services
-run.user_actions = run.user_actions[run.user_actions['User'].isin(run.users['User'].tolist()+[-1])]
-run.user_actions = run.user_actions[run.user_actions['Source_Service'].isin(run.services['Service'].tolist()+[-1])]
-run.user_actions = run.user_actions[run.user_actions['Target_Service'].isin(run.services['Service'].tolist()+[-1])]
+#run.user_actions = run.user_actions[run.user_actions['User'].isin(run.users['User'].tolist()+[-1])]
```

Comment:
We should talk about these:
1. Do we consider metrics valid with a different input dataset, since other metrics use another subset of user_actions/recommendations? Even for that not often cases such as Click-Through Rate?
2. If yes, should we consider adding a filter function as a decorator, to enable it when necessary?